src/frontend/js/components/ReferralTable/EnabledFiltersList.tsx

```diff
@@ -0,0 +1,308 @@
+import React, { Dispatch, Fragment, SetStateAction } from 'react';
+import { defineMessages, FormattedMessage } from 'react-intl';
+import { useQueries, UseQueryResult } from 'react-query';
+import { useUIDSeed } from 'react-uid';
+
+import { appData } from 'appData';
+import { GenericErrorMessage } from 'components/GenericErrorMessage';
+import { Spinner } from 'components/Spinner';
+import { fetchList } from 'data/fetchList';
+import { APIList, TopicLite, UnitLite, UserLite } from 'types';
+import { referralStateMessages } from 'utils/sharedMessages';
+import { getUserFullname } from 'utils/user';
+import { sharedMessages } from './sharedMessages';
+import { FilterColumns, FiltersDict } from './types';
+
+const messages = defineMessages({
+  dueDateFilter: {
+    defaultMessage:
+      'Between {due_date_after, date, short} and {due_date_before , date, short}',
+    description: 'Text for the filter label for due date filtering.',
+    id: 'components.ReferralTable.EnabledFiltersList.dueDateFilter',
+  },
+  loadingActiveFilters: {
+    defaultMessage: 'Loading active filters...',
+    description:
+      'Accessible alt text for the spinner while loading human readable filter values.',
+    id: 'components.ReferralTable.EnabledFiltersList.loadingActiveFilters',
+  },
+  removeFilter: {
+    defaultMessage: 'Remove filter',
+    description:
+      'Accessible title for the button to remove the related filter.',
+    id: 'components.ReferralTable.EnabledFiltersList.removeFilter',
+  },
+});
+
+interface EnabledFiltersListProps {
+  filters: FiltersDict;
+  setFilters: Dispatch<SetStateAction<FiltersDict>>;
+}
+
+export const EnabledFiltersList = ({
+  filters,
+  setFilters,
```

Comment:
injecting setState (here setFilters) into a child component is not a good practice in React at least for reusability. It's better to add a function, declared in the parent (like updateFilters() for example that will just setFilters. 

You can let it like that it doesn't really matter.

Reply:
I agree with this. Reusability is not my concern here but I'm thinking of using `useReducer` instead to make it more declarative in components and let the reducer hook implement the filter state logic.

However I'd rather do it in a separate refactoring PR as it touches other places where filters are managed.